### PR TITLE
Fix checkout with shipping and digital product

### LINKED ISSUE #5006: unable to checkout on graphql mutation [digital product]
### What I'm trying to achieve
checkout error on when only digital product is left in the cart system, if there is shipping method assigned before

### Steps to reproduce the problem
1. add shippable product
2. add non-shippable (digital) product
4. add shipping method
3. remove all shippable product, left only non-shippable product
4. add shippable product again
5. Error "This checkout doesn't need shipping" caused by execution of graphql mutation `updateCheckoutLine`

### What I expected to happen
able to add shippable product to the checkout cart
suspected: shipping method is not removed when only digital product is left on the cart

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because resolve #5006 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 2312804e35213377dd3235e7e5e5bcd8d1ee5b19 (mirumee/saleor @ 5006/fix_checkout_with_shipping_and_digital_product)
Base comparison is da715b1be241a8f456221a7771ec56491e3d3f9f.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         10	         10	              0
  complete checkout                          	          8	          8	              0
  create checkout                            	         50	         50	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	          9	          9	              0

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/graphql/checkout/mutations.py [hunk 1]
```diff
@@ -85,9 +85,14 @@ def update_checkout_shipping_method_if_invalid(checkout: models.Checkout, discou
...
 
-    is_valid = clean_shipping_method(
-        checkout=checkout, method=checkout.shipping_method, discounts=discounts
-    )
+    is_valid = True
+    try:
+        is_valid = clean_shipping_method(
+            checkout=checkout, method=checkout.shipping_method, discounts=discounts
+        )
+    except ValidationError:
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations.py]: Not 100% sure about this - all `ValidationError`s we raise in Graphene are supposed to be returned as errors in API. If we're catching it here, it means that maybe we shouldn't raise it in `clean_shipping_method`?  Maybe resetting the shipping method should happen in `update_checkout_shipping_method_if_invalid`?

Comment (fowczarek) [reply saleor/graphql/checkout/mutations.py]: How about the current solution?

File: saleor/graphql/checkout/mutations.py [hunk 2]
```diff
@@ -335,7 +335,7 @@ def perform_mutation(cls, _root, info, checkout_id, lines, replace=False):
         quantities = [line.get("quantity") for line in lines]
 
         check_lines_quantity(variants, quantities)
-        update_checkout_shipping_method_if_invalid(checkout, info.context.discounts)
+        # update_checkout_shipping_method_if_invalid(checkout, info.context.discounts)
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations.py]: What about this line?

**Pull request merged**